### DISCUSSION TITLE
Nextjs Cpanel Hosting  Issue

### DISCUSSION THREAD

**Discussion by hadiswaleh** [vercel/next.js#54593] [App Router]
Summary
Hi,

i need help as i am stuck on how to deploy next js app to cPanel hosting which supports Node.js applications.
Here is what i have

Next.js app which uses next app router.
Next js based API routes.
Using MySQL database for data api.
App is working in localhost
Here is what i want

I have a A2 Hosting with Cpanel
it supports Node.js applications
I have MySQL Database
i want to deploy my next.js app on this hosting
what is the best approach to do this? without losing + points of Next.js?

https://youtu.be/lex3qZAf_Ok?si=Da3rkiYUjIIuTrnN
Followed this steps also but no solution. Page open blank on live url.

**Comment by mailtoabusadat** [comment]
Same problem with me

**Reply by hadiswaleh** [reply]
Did you check any other forum for solution? 
Nextjs is giving headaches 😂 specially for cpanel

**Reply by mailtoabusadat** [reply]
I have searched variously forum / Youtube / Stack Overflow and google about this issue and have not found any solution.

**Reply by hadiswaleh** [reply]
I exported static files as html and deployed on cpanel.
This was the only last solution.
